aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -1835,61 +1885,100 @@ Tensor sparse_compressed_to_sparse(const Tensor& self, c10::optional<c10::Layout
       return sparse_compressed_to_sparse_bsr(self, *blocksize, dense_dim_opt);
     } else {
       const auto blocksize_ = at::sparse_csr::getBlockSize(self);
-      TORCH_CHECK(blocksize_.size() == 2, "to_sparse: ", self.layout(), " to ", layout_,
-                  " conversion requires blocksize specified.");
       return sparse_compressed_to_sparse_bsr(self, blocksize_, dense_dim_opt);
     }
   case kSparseBsc:
     if (blocksize.has_value()) {
       return sparse_compressed_to_sparse_bsc(self, *blocksize, dense_dim_opt);
     } else {
       const auto blocksize_ = at::sparse_csr::getBlockSize(self);
-      TORCH_CHECK(blocksize_.size() == 2, "to_sparse: ", self.layout(), " to ", layout_,
-                  " conversion requires blocksize specified.");
       return sparse_compressed_to_sparse_bsc(self, blocksize_, dense_dim_opt);
     }
```

Comment:
Nit:
```suggestion
return sparse_compressed_to_sparse_bsc(self, blocksize.value_or(at::sparse_csr::getBlockSize(self)), dense_dim_opt);
```

Reply:
Changed.

Reply:
This actually depends on how `value_or` is implemented and whether the compiler is able to avoid evaluating the argument when it will be not used.

If not, another option is to use `auto blockssize_ = (blocksize.has_value() ? *blocksize: at::sparse_csr::getBlockSize(self))`. Recall, the point of the suggestion was to eliminate the if-block with repeated code.

Btw, if `self` does not have a block size, the exception ought to be raised anyway when the block size is not specified. If argument checking failed to raise the exception, it would be raised by getBlockSize.

Reply:
Yeah, it was changed to:
```C++
return sparse_compressed_to_sparse_bsc(
               self,
               (self.layout() == kSparseBsr) ?
                   blocksize.value_or(at::sparse_csr::getBlockSize(self)) :
                   *blocksize,
               dense_dim_opt);
```
(It's just that the PR is not updated yet, trying to gather mutliple changes together, in order not to trigger CI very often.)